assets/js/modules/analytics/datastore/report.js

```diff
@@ -92,7 +95,12 @@ const baseResolvers = {
 			return;
 		}
 
-		yield fetchGetReportStore.actions.fetchGetReport( options );
+		const { error } = yield fetchGetReportStore.actions.fetchGetReport( options );
+
+		if ( normalizeReportOptions( options ).metrics.some( ( { expression } ) => /^ga:adsense/.test( expression ) ) ) {
+			const isRestrictedMetricError = error?.code === 400 && error?.message.startsWith( 'Restricted metric' );
```

Comment:
This wasn't specified in the ACs, but seeing this I think it would be more accurate to check for any `ga:adsense*` in the error message as well. Otherwise, theoretically it could be that a request which includes an AdSense metric fails due to _another_ restricted metric, but would still (incorrectly) raise this flag of AdSense not being linked.